### DISCUSSION TITLE
Migrating from XAMARIN TO MAUI APP:

### DISCUSSION THREAD

**Discussion by AngeloLazzari** [PrismLibrary/Prism#3215] [.NET MAUI] [answered]
Hi, we are migrating a xamarin app to Maui, and, following your documentation and videos, we installed this Prism packages:
`		<PackageReference Include="Prism.Container.DryIoc" Version="9.0.88" />
		<PackageReference Include="Prism.Core" Version="9.0.401-pre" />
		<PackageReference Include="Prism.Maui" Version="9.0.401-pre" />`
but:
a) we cannot find the .UsePrismApp on the MauiProgram.cs, I mean, I cannot find which "using Prism..." I need to put in my "using" list..
b) in the App.xaml.cs we used in Xamarin the  var result = await NavigationService.NavigateAsync("/PaginaLogin");  from Prism.Navigation but in the new MAUI project is not resolving it...

we even checked your full navigation repo but seeing the code we see that you don't use the .userPrismApp but you use the .UsePrims .. but visual Code gives me the same error I have with my app ... 

Can you please help me?
thank you very much

**Comment by dansiegel** [comment, marked as answer]
You have every reference you don't need and not the one that you do need.

You ONLY need a reference to `Prism.DryIoc.Maui`

**Reply by AngeloLazzari** [reply]
Thank for the quick reply. I uninstalled the Prism.Core and Prism.Maui nugets and left the Prism.Container.DryIoc ( this last one because if I look for Prism.DryIoc.Maui nuget only the Prism.Container.DryIoc is displayed... ... but nothing has changed.. 

Can you give a little help with that please? Thank you so much!

**Reply by AngeloLazzari** [reply]
Sorry, I founf the Prism.DryIoc.Maui as prerelease ... I tell you how it will work thanks....

**Reply by AngeloLazzari** [reply]
Ok, I got it, now the project is detecting the UsePrism.

So, based on your documentation [here ](https://prismlibrary.github.io/docs/maui/migrating.html) we were using in our XAMARIN App.xaml.cs both the RegisterTypes and OnInitialized methods, so we need to move them directly into the .UsePrism step of the Mauibuilder... correct? 

I ask you this because xamarin OnInitialized has a NavigationService.NavigateAsync that i should put into the .CreateWindow(async navigationService => ... correct? 

Sorry for this questions but I really need to make the app working thanks

**Reply by AngeloLazzari** [reply]
thank you very much!
